python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/ethernet-interfaces.j2

```diff
@@ -45,19 +45,41 @@
 {%             endif %}
 {%         elif ethernet_interface.type | arista.avd.default('switched') == 'switched' %}
 {%             set description = ethernet_interface.description | arista.avd.default("-") %}
-{%             set mode = ethernet_interface.mode | arista.avd.default("access") %}
-{%             set vlans = ethernet_interface.vlans | arista.avd.default('-') %}
-{%             if  ethernet_interface.native_vlan_tag is arista.avd.defined(true) %}
+{%             if  ethernet_interface.switchport.mode is arista.avd.defined %}
+{%                 set mode = ethernet_interface.switchport.mode %}
+{%                 if mode in ["access", "dot1q-tunnel"] %}
+{%                     set vlans = ethernet_interface.switchport.access_vlan %}
+{%                 elif mode == "trunk" %}
+{%                     set vlans = ethernet_interface.switchport.trunk.allowed_vlan %}
+{%                 elif mode == "trunk phone" %}
+{%                     set vlans = ethernet_interface.switchport.phone.vlan %}
+{%                 endif %}
```

Comment:
We do not want logic evaluating `mode` here. Just check for each key and output it in the table if set.

Reply:
updated!